src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -619,6 +622,30 @@ class spell_pri_benediction : public SpellScript
     }
 };
 
+// 215768 - Blaze of Light
+class spell_pri_blaze_of_light : public AuraScript
+{
+    void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* caster = eventInfo.GetActor();
+        Unit* target = eventInfo.GetProcTarget();
+        if (!caster || !target)
+            return;
+
+        if (caster->IsValidAttackTarget(target))
+        {
+            caster->CastSpell(target, SPELL_PRIEST_BLAZE_OF_LIGHT_DECREASE, aurEff);
+            return;
```

Comment:
is there any reason you're returning here?

Reply:
theres no point since i check the target, right?

Reply:
```c
if (caster->IsValidAttackTarget(target))
    caster->CastSpell(target, SPELL_PRIEST_BLAZE_OF_LIGHT_DECREASE, aurEff);
else
    caster->CastSpell(target, SPELL_PRIEST_BLAZE_OF_LIGHT_INCREASE, aurEff);
```

Reply:
i see i see